sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeExtractors.scala

```diff
@@ -275,9 +275,9 @@ trait GetArrayItemUtil {
     if (ordinal.foldable && !ordinal.nullable) {
       val intOrdinal = ordinal.eval().asInstanceOf[Number].intValue()
       child match {
-        case CreateArray(ar) if intOrdinal < ar.length =>
+        case CreateArray(ar, emptyCollection) if intOrdinal < ar.length =>
```

Comment:
Is emptyCollection not used, right? You can add unapply to `CreateArray` and avoid modification of this place.

Reply:
You can just write `case CreateArray(ar, _)`